OpenRA.Mods.Common/Activities/RepairBuilding.cs

```diff
@@ -32,11 +33,12 @@ protected override bool TryStartEnter(Actor self, Actor targetActor)
 		{
 			enterActor = targetActor;
 			enterHealth = targetActor.TraitOrDefault<IHealth>();
+			enterEngineerRepariable = targetActor.TraitOrDefault<EngineerRepairable>();
 
 			// Make sure we can still repair the target before entering
 			// (but not before, because this may stop the actor in the middle of nowhere)
 			var stance = self.Owner.Stances[enterActor.Owner];
-			if (enterHealth == null || enterHealth.DamageState == DamageState.Undamaged || !info.ValidStances.HasStance(stance))
+			if (enterHealth == null || enterHealth.DamageState == DamageState.Undamaged || enterEngineerRepariable == null || enterEngineerRepariable.IsTraitDisabled || !info.ValidStances.HasStance(stance))
```

Comment:
It would look a little cleaner to use `Exts.IsTraitEnabled(enterEngineerRepariable)` instead of `enterEngineerRepariable == null || enterEngineerRepariable.IsTraitDisabled`.

Reply:
I'm not sure if i'm reading the code wrong, `Exts.IsTraitEnabled(enterEngineerRepariable)` seems to be `enterEngineerRepariable == null || !enterEngineerRepariable.IsTraitDisabled`, not `enterEngineerRepariable == null || enterEngineerRepariable.IsTraitDisabled`. Tho, i feel like a sensible IsEnabled check should be `not null and not disabled` not `null or not disabled`. So i'm not really sure.

Reply:
Correction, you're right. I looked for other code with that pattern and didn't find any. That is why there isn't a  Exts.IsTraitDisabled, the correct function if it existed, like I was sure there was.